### PR TITLE
Consistent Plot Color Schemes

### PR DESCRIPTION
**Pull request by wbinventor**
This PR adds a new `colorize(geometry, seed=1)` routine to the Python API's `Plot` and `PlotsFile` objects. The `Plot.colorize(...)` routine generates a random (but reproducible per the RN seed) color scheme based on the IDs of *all* materials or cells in the geometry. The `PlotsFile.colorize(...)` generates a consistent color scheme for all plots so that the color used for a cell or material will appear the same regardless of the perspective (*e.g.*, plot basis or width).

### CODE REVIEW

File: openmc/plots.py [hunk 1]
```diff
@@ -261,6 +261,43 @@ def __repr__(self):
...
+
+        # Set the seed for the random number generator
+        np.random.seed(seed)
+
+        # Generate random colors for each feature
+        self.col_spec = {}
+        for domain in domains:
+            r = np.random.randint(0, 255)
+            g = np.random.randint(0, 255)
+            b = np.random.randint(0, 255)
```
Comment (paulromano) [inline openmc/plots.py]: `np.random.randint(a, b)` produces random integers in the range [a,b), so you probably want b=256, yeah?

Comment (paulromano) [inline openmc/plots.py]: Also, curious to know if you have a specific rationale for using `np.random.randint` over the standard library `random.randint`

File: openmc/plots.py [hunk 2]
```diff
@@ -261,6 +261,43 @@ def __repr__(self):
         string += '{0: <16}{1}{2}\n'.format('\tCol Spec', '=\t', self._col_spec)
         return string
 
+    def colorize(self, geometry, seed=1):
+        """Generate a color scheme for each domain in the plot.
+
+        This routine may be used to generate random, reproducible color schemes.
+        The colors generated are based upon cell/material IDs in the geometry.
+
+        Params
```
Comment (paulromano) [inline openmc/plots.py]: Parameters

File: openmc/plots.py [hunk 3]
```diff
@@ -349,6 +386,25 @@ def remove_plot(self, plot):
...
         self._plots.remove(plot)
 
+    def colorize(self, geometry, seed=1):
+        """Generate a consistent color scheme for each domain in each plot.
+
+        This routine may be used to generate random, reproducible color schemes.
+        The colors generated are based upon cell/material IDs in the geometry.
+        The color schemes will be consistent for all plots in "plots.xml".
+
+        Params
```
Comment (paulromano) [inline openmc/plots.py]: Parameters

File: openmc/plots.py (hunk 1 above)
Comment (wbinventor) [inline openmc/plots.py]: Very good call on the upper bound. No particular rationale other than that `numpy` was already imported and `random` would be a new import. I can switch if you want.

**Comment by wbinventor** [comment]
All comments have been addressed

**Pull request merged**